client/signup/config/flows.js

```diff
@@ -133,6 +133,16 @@ function getChecklistThemeDestination( { flowName, siteSlug, themeParameter } )
 	return `/home/${ siteSlug }`;
 }
 
+function getWithThemeDestination( { siteSlug, themeParameter, styleVariation, themeType } ) {
+	if ( 'dot-org' === themeType ) {
+		return `/marketplace/theme/${ themeParameter }/install/${ siteSlug }`;
+	}
+
+	const style = styleVariation ? `&style=${ styleVariation }` : '';
+
+	return `/setup/site-setup/designSetup?siteSlug=${ siteSlug }&theme=${ themeParameter }${ style }&hideBack=true`;
```

Comment:
What do you think about excluding free themes here? Users don't need to perform any additional step to activate them, so adding one extra step could affect the signup process.